rand_distr/src/utils.rs

```diff
@@ -200,33 +25,33 @@ impl Float for f64 {
 /// `Ag(z)` is an infinite series with coefficients that can be calculated
 /// ahead of time - we use just the first 6 terms, which is good enough
 /// for most purposes.
-pub(crate) fn log_gamma(x: f64) -> f64 {
+pub(crate) fn log_gamma<F: num_traits::Float>(x: F) -> F {
```

Comment:
How accurate is the single-precision version of this function?

Reply:
I am not sure, but at the first glance we should not encounter serious numerical instability issues. Using `f64` under the hood even though user requests `f32` output may have serious performance implications on hardware which does not support 64-bit floats, so I think `log_gamma` should be generic over float types.

Reply:
Agreed, it doesn't look like there are any serious issues.

I don't see much incentive to care about performance on platforms without a 64-bit FPU without an interested party at least supplying benchmarks. Additionally it may enable a little AVX vectorisation where the constants are folded in, if code were re-written for it, but benefits would likely be small.

I guess we can accept this change as-is.